packages/utils/src/diff/index.ts

```diff
@@ -198,6 +211,21 @@ function getObjectsDifference(
   if (aCompare === bCompare) {
     return getCommonMessage(NO_DIFF_MESSAGE, options)
   }
+  else if (options?.expected) {
+    const clonedA = deepClone(a, { forceWritable: true })
+    const clonedB = deepClone(b, { forceWritable: true })
```

Comment:
What are the performance implications of this? It's the second time we deeply cloned the object, and in `replaceDiffData` we only iterate the first level

Reply:
This was intended to reduce unexpected side effects, but since those values are not included in the return value, it would be better to remove them as you suggested.